skelerest/test/test_rest_request.py

```diff
@@ -0,0 +1,128 @@
+import unittest
+from schema import SchemaError
+from ..rest_request import RestRequest
+
+class TestRestRequest(unittest.TestCase):
+
+    CONFIG_VALID = {
+        "name": "test-project",
+        "endpoint": "http://not a real {site}",
+        "method": "POST",
+        "params": [
+            {"name": "one", "value": "{param-one:1}"},
+            {"name": "two", "value": "{param-two:2}"}
+        ],
+        "headers": [
+            {"name": "a", "value": "{header-one:A}"},
+            {"name": "b", "value": "{header-two:B}"}
+        ],
+        "body": {
+            "id": "{id:0}",
+            "name": "test",
+            "items": ["a", "b", "c"],
+            "parent": {
+                "id": "{parent-id}",
+                "name": "{parent-name}"
+            }
+        }
+    }
+
+    CONFIG_INVALID = {
+        "endpoint": 0,
+        "method": 123,
+        "params": "test",
+        "headers": "test",
+        "body": 123
+    }
+
+    def test_load(self):
+        restRequest = RestRequest.load(self.CONFIG_VALID)
+
+        self.assertEqual(restRequest.name, "test-project")
+        self.assertEqual(restRequest.endpoint, "http://not a real {site}")
+        self.assertEqual(restRequest.method, "POST")
+
+        i = 0
+        exp_param_names = ["one", "two"]
+        exp_param_values = ["{param-one:1}", "{param-two:2}"]
+        for param in restRequest.params:
+            self.assertEqual(param.name, exp_param_names[i])
+            self.assertEqual(param.value, exp_param_values[i])
+            i += 1
```

Comment:
You can use enumerate to manage the `i` counter:
```suggestion
        exp_param_names = ["one", "two"]
        exp_param_values = ["{param-one:1}", "{param-two:2}"]
        for i, param in enumerate(restRequest.params):
            self.assertEqual(param.name, exp_param_names[i])
            self.assertEqual(param.value, exp_param_values[i])
```